rp2040-hal/src/gpio/pin.rs

```diff
@@ -959,9 +959,13 @@ macro_rules! gpio {
                 impl Pins {
                     /// Take ownership of the PAC peripherals and SIO slice and split it into discrete [`Pin`]s
                     pub fn new(io : [<IO_ $Group:upper>], pads: [<PADS_ $Group:upper>], sio: [<SioGpio $Group>], reset : &mut pac::RESETS) -> Self {
+                        pads.reset_bring_down(reset);
+                        io.reset_bring_down(reset);
```

Comment:
I still think there should be a way to initialize HAL devices without a full peripheral reset, especially for GPIO where reconfigurations could introduce unnecessary glitches.
However, as a default, those resets are correct.
Special, potentially unsafe, alternatives to create those objects without resets can be added later, if actually needed.

Reply:
I do think so too. As discussed over matrix, a solution could be to introduce an extra state for the peripherals and let the user choose between the safe (reset) way or the unsafe shortcut to the state they want. (In some cases we could just use things like DynPin an allow to move into a specific state).

But in the mean time i think we should address that rather important inconsistency as more using will be confused by the result they get from resetting/reflashing their development target.